app/stac_api/models.py

```diff
@@ -197,27 +190,469 @@ def update_geoadmin_variants(self, asset_geoadmin_variant, asset_proj_epsg, asse
                 "Error when updating collection's summaries values due to asset update."
             ))
 
-    def update_temporal_extent(self, item_properties_start_datetime, item_properties_end_datetime):
+    def insert_item(
+        self, old_start_datetime, new_start_datetime, old_end_datetime, new_end_datetime, item_id
+    ):
```

Comment:
I would removed unused parameter which don't make sense in this method, `old_*`. And aslo rename the method, we don't insert a item in collection here we update the temporal extent due to item insertion: `update_temporal_extent_insert` it is a bit a long name but much more clearer to the reader in my opinion

Reply:
done